extensions/amp-script/0.1/test/unit/test-amp-script.js

```diff
@@ -294,6 +294,48 @@ describes.repeated(
             return promise.should.be.rejected;
           });
         });
+
+        describe('callFunction', () => {
+          let script;
+          let service;
+          let element;
+
+          beforeEach(() => {
+            // Sets up <script>, <amp-script>, and the AmpScriptService.
+            const local = document.createElement('script');
+            local.setAttribute('id', 'local-script');
+            local.setAttribute('target', 'amp-script');
+            env.ampdoc.getBody().appendChild(local);
+
+            element = document.createElement('amp-script');
+            env.ampdoc.getBody().appendChild(element);
+            script = new AmpScript(element);
+            element.getImpl = () => Promise.resolve(script);
+            script.getAmpDoc = () => env.ampdoc;
+            service = new AmpScriptService(env.ampdoc);
+            script.setService(service);
+          });
+
+          it('should throw if there is no amp-script with the proper target', async () => {
+            const call = () =>
+              service.callFunction('local-script', 'fetchData');
+            const errorMessage = /could not find amp-script with/;
+            expectAsyncConsoleError(errorMessage);
+            expect(call).to.throw(errorMessage);
+          });
+
+          it('Waits for initialization to complete before returning', async () => {
+            element.setAttribute('script', 'local-script');
+            const result = service.callFunction('local-script', 'fetchData');
```

Comment:
Is testing for bad inputs handled in worker dom? Also can that call an identifier with dots?

Reply:
I think I answer that in a response to an earlier question, let me know if I missed something! 
https://github.com/ampproject/amphtml/pull/29329#discussion_r457572600

Reply:
I may have missed if there are test cases for this in worker dom, but I mean strings that are empty or otherwise not valid js variable names, i.e. `123` or anything containing odd symbols.

Reply:
We don't run the function name they pass directly into an `eval()` or anything like that. `worker-dom` exposes a function called `exportFunction(string, handle)`. As long as the fnIdent they pass in matches a string that was exported it'll work.

That is an interesting point that technically invalid identifiers can pass through this unharmed, I'm not convinced thats a bad thing.